examples/amm/src/service.rs

```diff
@@ -25,7 +25,7 @@ impl Service for Amm {
 
     async fn handle_query(
         self: Arc<Self>,
-        _context: &QueryContext,
+        _runtime: &mut ServiceRuntime,
```

Comment:
Should it be `&ServiceRuntime`?

Reply:
Not really, because we're using the same caching mechanism as the `ContractRuntime`.

Reply:
Nevermind, I experimented with using a `Cell` and it works quite nicely :tada: